packages/fetch-and-compile/lib/multiple.ts

```diff
@@ -0,0 +1,60 @@
+import debugModule from "debug";
+const debug = debugModule("fetch-and-compile:multiple");
+import type {
+  Recognizer,
+  FailureType,
+  FetchAndCompileResult,
+  FetchAndCompileMultipleResult,
+  FetchAndCompileFailureRecord
+} from "./types";
+
+export class MultipleRecognizer implements Recognizer {
+  private unrecognizedAddresses: string[];
+  private addressesToSkip: Set<string> = new Set();
+  private results: { [address: string]: FetchAndCompileResult } = {};
+  private failureLog: { [address: string]: FetchAndCompileFailureRecord } = {};
+
+  constructor(addresses: string[]) {
+    this.unrecognizedAddresses = [...new Set(addresses)]; //remove duplicates and clone
```

Comment:
Would the input include checksum'd addresses?

Reply:
Oh, good question.  Yeah, I had been assuming all inputs would be checksummed, but I guess that's not necessarily the case, is it?  Perhaps I should either checksum everything or lowercase everything to better ensure that addresses don't occur twice?

Reply:
Either would be fine, though it'd be faster and less logic to normalize by lowercase.

Reply:
Cool, I'll do that then.